File: src/main/java/com/damianryan/octopus/model/Property.java
package com.damianryan.octopus.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;
import org.springframework.lang.Nullable;

import java.time.Instant;
import java.util.List;

@Data
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Property {

    int id;

    @JsonProperty("moved_in_at")
    Instant movedInAt;

    @JsonProperty("moved_out_at")
    @Nullable
    Instant movedOutAt;

    @JsonProperty("address_line_1")
    String addressLine1;

    @JsonProperty("address_line_2")
    String addressLine2;

    @JsonProperty("address_line_3")
    String addressLine3;

    String town;

    String county;

    String postcode;

    @JsonProperty("electricity_meter_points")
    List<ElectricityMeterPoint> electricityMeterPoints;

    @JsonProperty("gas_meter_points")
    List<GasMeterPoint> gasMeterPoints;
}

File: src/main/kotlin/com/damianryan/octopus/model/Property.kt
package com.damianryan.octopus.model

import com.fasterxml.jackson.annotation.JsonProperty
import java.time.Instant

data class Property(
    val id: Int = 0,
    @JsonProperty("moved_in_at") val movedInAt: Instant? = null,
    @JsonProperty("moved_out_at") val movedOutAt: Instant? = null,
    @JsonProperty("address_line_1") val addressLine1: String? = null,
    @JsonProperty("address_line_2") val addressLine2: String? = null,
    @JsonProperty("address_line_3") val addressLine3: String? = null,
    val town: String? = null,
    val county: String? = null,
    val postcode: String? = null,
    @JsonProperty("electricity_meter_points") val electricityMeterPoints: List<ElectricityMeterPoint>? = null,
    @JsonProperty("gas_meter_points") val gasMeterPoints: List<GasMeterPoint>? = null,
)